test/index.js

```diff
@@ -1914,6 +1914,172 @@ describe('control', function() {
 
   });
 
+  describe('allPass', function() {
+
+    it('is a binary function', function() {
+      eq(typeof S.allPass, 'function');
+      eq(S.allPass.length, 2);
+    });
+
+    it('type checks its arguments', function() {
+      assert.throws(function() { S.allPass('wrong'); },
+             errorEq(TypeError,
+                     '‘allPass’ requires a value of type Array ' +
+                     'as its first argument; received "wrong"'));
+    });
+
+    it('returns true when given an empty array of predicates', function() {
+      eq(S.allPass([], 'quiessence'), true);
+    });
+
+    it('returns true when all predicates pass', function() {
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                    'quiessence'),
+         true);
+    });
+
+    it('returns false when some predicates fail', function() {
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                    'quintessential'),
+         false);
+
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                    'incandescent'),
+         false);
+
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                    'fissiparous'),
+         false);
+    });
+
+    it('returns false when all predicates fail', function() {
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                    'empathy'),
+         false);
+    });
+
+    it('short-circuits when one predicate fails', function() {
+      var evaluated = false;
+      var evaluate = function() { evaluated = true; };
+
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    evaluate],
+                    'incandescent'),
+         false);
+      eq(evaluated, false);
+    });
+
+    it('is curried', function() {
+      eq(S.allPass([]).length, 1);
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)])('quiessence'),
+         true);
```

Comment:
We could use a shorter expression here since we're just testing currying. Perhaps:

```javascript
eq(S.allPass([S.test(/q/)])('quiessence'), true);
```